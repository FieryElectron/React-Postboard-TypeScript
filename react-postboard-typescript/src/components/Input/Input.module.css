.inputContainer {
    margin: 5px;
    padding: 5px;
}


.input {
    border: 0;
    border-bottom: 2px solid gray;
    outline: 0;
    font-size: 1.5rem;
    color: black;
    padding: 7px 0;
    background: transparent;
    transition: border-color 0.2s;
}