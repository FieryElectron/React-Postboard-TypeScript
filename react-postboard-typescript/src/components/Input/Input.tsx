import styles from './Input.module.css';

type Props = {
    placeholder:string
    type?:string
    onChange?:React.ChangeEventHandler
}

const Input = ({ placeholder, type , onChange} : Props) => {
    return (
        <div data-testid='inputContainer' className={styles.inputContainer}>
            <input data-testid='input' className={styles.input} placeholder={placeholder} type={type} onChange={onChange} spellCheck='false'/>
        </div>
    )
}

Input.defaultProps = {
    text: '',
    placeholder: 'placeholder',
    type: 'text'
}

export default Input;