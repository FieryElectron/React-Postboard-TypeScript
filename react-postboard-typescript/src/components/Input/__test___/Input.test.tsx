import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import Input from '../Input';

let container: HTMLElement;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container.remove();
});

it('empty test', () => {

});

it('renders Input without crashing', () => {
  act(() => {
    render(<Input/>);
  });
});

it('renders Input class correctly', () => {
  act(() => {
    render(<Input />);
  });

  const inputContainer = document.querySelector('[data-testid=inputContainer]')!;
  const input = document.querySelector('[data-testid=input]')!;

  expect(inputContainer).toHaveAttribute('class', 'inputContainer');
  expect(input).toHaveAttribute('class', 'input');
});

it('renders Input placeholder correctly', () => {
  act(() => {
    render(<Input placeholder='Input some text' />);
  });

  const input = document.querySelector('[data-testid=input]')!;
  expect(input).toHaveAttribute('placeholder', 'Input some text');
});

it('renders Input type correctly', () => {
  act(() => {
    render(<Input type='password' />);
  });

  const input = document.querySelector('[data-testid=input]')!;
  expect(input).toHaveAttribute('type', 'password');
});

it('should call onChange callback', () => {
  const onChange = jest.fn();
  act(() => {
    render(<Input onChange={onChange} />);
  });
  const input = document.querySelector('[data-testid=input]')!;
  fireEvent.change(input, { target: { value: 'some text' } });
  expect(onChange).toHaveBeenCalled();
});

it('matchs snapshot 1', () => {
  const onChange = jest.fn();
  const tree = renderer.create(<Input placeholder='placeholder' type='password' onChange={onChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('matchs snapshot 2', () => {
  const onChange = jest.fn();
  const tree = renderer.create(<Input placeholder='username' onChange={onChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});

