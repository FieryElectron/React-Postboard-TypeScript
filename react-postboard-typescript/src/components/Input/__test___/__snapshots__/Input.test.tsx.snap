// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `
<div
  className="inputContainer"
  data-testid="inputContainer"
>
  <input
    className="input"
    data-testid="input"
    onChange={[MockFunction]}
    placeholder="placeholder"
    spellCheck="false"
    type="password"
  />
</div>
`;

exports[`matchs snapshot 2 1`] = `
<div
  className="inputContainer"
  data-testid="inputContainer"
>
  <input
    className="input"
    data-testid="input"
    onChange={[MockFunction]}
    placeholder="username"
    spellCheck="false"
    type="text"
  />
</div>
`;
